src/cli.cr

```diff
@@ -119,7 +119,7 @@ module Shards
     {% else %}
       shards_opts = ENV.fetch("SHARDS_OPTS", "").split
     {% end %}
-    ARGV.concat(shards_opts)
+    ARGV.dup.concat(shards_opts)
```

Comment:
**question:** I'm wondering why is this `dup` necessary?

Reply:
`cli_options` modifies `ARGV` by concatenating `SHARDS_OPTS` possible values to it. Then OptionParser consumes that by eliminating the processed arguments.

This means that multiple calls to `cli_options` will return different arguments before and after OptionParser. This breaks attempting to pass arguments to the subcommand.

Duping ARGV ensures `cli_options` returns a copy of the parameters and keeps ARGV clean from the concatenation with `SHARDS_OPTS`.

Reply:
Makes sense. It's a bit convoluted I guess, but it's what we have to work with 🤷

Reply:
Indeed, but I didn't want to refactor the entire thing right now 😅